### PR TITLE
AppleSignInProperties structure

### PR DESCRIPTION
**Pull request by andrey-kashcheev**
First step on Apple Sign-In support

Relates-To: OLPEDGE-2482
Signed-off-by: Andrey Kashcheev <ext-andrey.kashcheev@here.com>

### CODE REVIEW

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 1]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
     unsigned int expires_in{0};
   };
 
+  /**
+   * @brief The Apple sign-in properties structure.
+   */
+  struct AppleSignInProperties {
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:154]: Let's change this awful approach and not do a nested struct. Please move to a separate file.

**Review by andescu — COMMENTED** [review]
Please enhance the commit message. Describe what you are adding and what it will be used for and why do we add it.

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 2]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
     unsigned int expires_in{0};
   };
 
+  /**
+   * @brief The Apple sign-in properties structure.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:152]:
```suggestion
   * @brief The structure of the Apple sign-in properties.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 3]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+   */
+  struct AppleSignInProperties {
+    /**
+     * @brief The Apple id_token to be used as bearer in the authorization
+     * request header.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:157]:
```suggestion
     * @brief The Apple ID token to be used as a bearer in the authorization
     * request header.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 4]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     */
+    std::string access_token;
+
+    /**
+     * @brief OIDC name property for the user.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:162]: Can we put it in the following way?
```suggestion
     * @brief Your OIDC name.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 5]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     *
+     * Required for the first time sign-in; optional for the subsequent
+     * sign-ins.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:165]:
```suggestion
     * Required for the first-time sign-in; optional for the subsequent
     * sign-ins.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 6]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     */
+    std::string name;
+
+    /**
+     * @brief The code of the country in which you live
+     * in the ISO 3166-1 alpha-3 format.
+     *
+     * Required for the first time sign-in; optional for the subsequent
+     * sign-ins.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:174]:
```suggestion
     * Required for the first-time sign-in; optional for the subsequent
     * sign-ins.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 7]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     */
+    std::string country_code;
+
+    /**
+     * @brief The code of the language that you speak
+     * in the ISO 639-1 alpha-2 format.
+     *
+     * Required for the first time sign-in; optional for the subsequent
+     * sign-ins.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:183]:
```suggestion
     * Required for the first-time sign-in; optional for the subsequent
     * sign-ins.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 8]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     */
+    std::string language;
+
+    /**
+     * @brief The application id.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:188]:
```suggestion
     * @brief The application ID.
```

File: olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h [hunk 9]
```diff
@@ -148,6 +148,54 @@ class AUTHENTICATION_API AuthenticationClient {
...
+     */
+    std::string client_id;
+
+    /**
+     * @brief The Platform realm the above client_id resp. application belongs
+     * to.
```
Comment (HalynaDumych) [inline olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationClient.h:194]:
```suggestion
     * @brief The platform realm to which the application ID (`client_id`) belongs.
```

**Review by HalynaDumych — CHANGES_REQUESTED** [review]

**Review by HalynaDumych — DISMISSED** [review]

File: olp-cpp-sdk-authentication/include/olp/authentication/AppleSignInProperties.h [hunk 10]
```diff
@@ -0,0 +1,163 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include "AuthenticationApi.h"
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/AppleSignInProperties.h:22]: Please include it with global includes, means <olp/authentication/AuthenticationApi.h>

File: olp-cpp-sdk-authentication/include/olp/authentication/AppleSignInProperties.h [hunk 11]
```diff
@@ -0,0 +1,163 @@
...
+   * @brief Gets the bearer in the authorization request header.
+   *
+   * @return The access token.
+   */
+  const std::string& GetAccessToken() const { return access_token_; }
+
+  /**
+   * @brief Sets the bearer in the authorization request header.
+   *
+   * @note The 'id_token' received after authorization should be used here.
```
Comment (andescu) [inline olp-cpp-sdk-authentication/include/olp/authentication/AppleSignInProperties.h:42]: Is this correct? No mention that this is an Apple id_token taken from Apple authentication. Please adapt this and the above GetAccessToken() Doxygen to mention that we are talking about a Apple token here.

**Review by andescu — APPROVED** [review]

**Review by kzvieriev — APPROVED** [review]

**Pull request merged**